src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSource.java

```diff
@@ -164,16 +167,22 @@ public ReadsSparkSource(final JavaSparkContext ctx, final ValidationStringency v
      * @return the header for the bam.
      */
     public SAMFileHeader getHeader(final String filePath, final GATKPathSpecifier referencePathSpecifier) {
+        final GATKPathSpecifier cramReferencePathSpec = checkCramReference(ctx, filePath, referencePathSpecifier);
+
         // GCS case
         if (BucketUtils.isGcsUrl(filePath)) {
-            try (ReadsDataSource readsDataSource = new ReadsDataSource(IOUtils.getPath(filePath))) {
-                return readsDataSource.getHeader();
+            final SamReaderFactory factory = SamReaderFactory.makeDefault().validationStringency(validationStringency);
+            try (ReadsDataSource readsDataSource = new ReadsDataSource(
+                    Collections.singletonList(IOUtils.getPath(filePath)),
+                    cramReferencePathSpec == null ?
```

Comment:
this is super nitpicky, but why not do this set or not as part of the factory initialization and then just always pass in factory?  it moves the ternary out of the new object declaration which seems preferable.

Reply:
Done.